flytekit/core/python_auto_container.py

```diff
@@ -294,12 +321,41 @@ def name(self) -> str:
     def load_task(self, loader_args: List[str]) -> PythonAutoContainerTask:
         _, entity_name, *_ = loader_args
         import gzip
+        import sys
 
         import cloudpickle
 
-        with gzip.open(PICKLE_FILE_PATH, "r") as f:
-            entity_dict = cloudpickle.load(f)
-        return entity_dict[entity_name]
+        try:
+            with gzip.open(PICKLE_FILE_PATH, "r") as f:
+                loaded_data = cloudpickle.load(f)
+        except TypeError:
+            raise RuntimeError(
+                "The Python version is smaller than the version used to create the pickle file. "
```

Comment:
nit: is the backwards compatibility a guarantee that cloudpickle gives? (As in, cloudpickle version X will always deserialize cloudpickle Y if X > Y). If that's not stated anywhere I'd just say that versions are different.

Reply:
No, I don't think `cloudpickle` has explicitly promised backward compatibility (but `pickle` does). I'll update that.